include/makefile.in

```diff
@@ -297,18 +297,27 @@ dist-devel-post: dist-devel-recursive
 dist-devel-recursive: dist-devel-pre
 
 stamp:
-	@cp stamp.h stamp2.h; \
-	stamp_version=\"$(PACKAGE_VERSION)\"; \
-	if [ -e $(rootfromhere)/.git ]; then \
-	  stamp_version=\"`$(GIT) describe --tag --always 2> /dev/null | sed -e s/-bootstrap$$//`\"; \
-	fi; \
-	if fgrep "#define ___STAMP_VERSION $$stamp_version" stamp.h > /dev/null; then \
-	  echo "*** include/stamp.h untouched (___STAMP_VERSION did not change)"; \
-	else \
-	  echo "*** include/stamp.h updated to have ___STAMP_VERSION=$$stamp_version"; \
-	  eval `date -u '+stamp_ymd=%Y%m%d stamp_hms=%H%M%S'`; \
-	  sed -e "s/#define ___STAMP_YMD [0-9][0-9]*/#define ___STAMP_YMD `expr 0 + $$stamp_ymd`/" -e "s/#define ___STAMP_HMS [0-9][0-9]*/#define ___STAMP_HMS `expr 0 + $$stamp_hms`/" -e "s/#define ___STAMP_VERSION .*/#define ___STAMP_VERSION $$stamp_version/" $(srcdirpfx)stamp.h > stamp2.h; \
-	  mv stamp2.h stamp.h; \
+	echo > stamp.h && \
```

Comment:
Sorry I didn't catch this before, but `stamp.h` should not be unconditionally modified if it does not have to change (e.g. if it was and will be an empty file, or if it was version V and will be version V).  This is to avoid recompiling files due to a needless change of file modification time.  The `stamp.h` file is particularly important to handle this way because all the runtime library .c files depend on `stamp.h` indirectly, so a change in this file causes all the runtime library to be recompiled.

Reply:
No problem. Is it adequate to write `stamp.h.new` as above, and then use `cmp` or `diff` to find out whether the contents changed?

Reply:
Wait, do we ever need to compare `stamp.h` to `stamp-release.h` or is it enough to save the previous `stamp.h` and compare the new `stamp.h` to it?

Reply:
The file `stamp-release.h` should not affect the generation of `stamp.h`.  Moreover it is only if `___STAMP_VERSION` needs to change that `stamp.h` needs to change.  The YMD and HMS indicate the last time it was noticed that `___STAMP_VERSION` changed.  If YMD and HMS were part of the comparison then `stamp.h` would always change.

Reply:
However if the YMD and HMS are generated from the commit date then a file comparison would work.

Reply:
BTW the same can be said for `stamp-release.h`.  Using the commit date would give a stamp that is consistent with the git history.

Reply:
OK, new commit using `grep` to see if the new `___STAMP_VERSION` can be found in the previous file.